python/pyspark/pandas/groupby.py

```diff
@@ -608,18 +608,18 @@ def max(self, numeric_only: Optional[bool] = False, min_count: int = -1) -> Fram
             min_count=min_count,
         )
 
-    def mean(self, numeric_only: Optional[bool] = True) -> FrameLike:
+    def mean(self, numeric_only: Optional[bool] = None) -> FrameLike:
         """
         Compute mean of groups, excluding missing values.
 
         Parameters
         ----------
-        numeric_only : bool, default True
+        numeric_only : bool, default None
             Include only float, int, boolean columns. If None, will attempt to use
             everything, then use only numeric data. False is not supported.
             This parameter is mainly for pandas compatibility.
 
-            .. versionadded:: 3.4.0
+            .. versionchanged:: 4.0.0
```

Comment:
shall we keep the old `versionchanged` and add this new one?
both `versionchanged` should doc what was changed.

Reply:
Makes sense to me. Updated!